tools/dm/skip-replace-sqls.md

```diff
@@ -0,0 +1,547 @@
+- If it is unacceptable in the actual production environment that the abnormal DDL statement is skipped in the downstream TiDB and it cannot be replaced with other DDL statements, then using this approach is not recommended.
+    - For example: `DROP PRIMARY KEY`
+    - In this scenario, you can only create a new table in the downstream with the new table schema (after executing the DDL statement), and re-import all the data into this new table. 
+- A single skip/replace operation targets at a single binlog event.
+- `--sharding` is only used to preset the operation to the sharding group. You must preset it before executing the DDL statement and presetting it after executing the DDL is not allowed.
+    - `--sharding` only supports presetting operations, and in this mode, you can only use `--sql-pattern` to match the binlog event.
+    - For the principles of replicating sharding DDL statements using DM, see [Merge and replicate data from sharded tables](/tools/dm/shard-merge.md#principles)
+
+#### Match the binlog event
+
+When the replication task gets interrupted because of the SQL execution error, you can obtain the position of the corresponding binlog event by using `query-error`. When you execute `sql-skip` or `sql-replace`, you can specify the position to match the binlog event.
+
+However, when you try to avoid breaking the replication by actively handling unsupported SQL statements, you cannot know in advance the position of the binlog event, so you need another approach to match the subsequent binlog events.
+
+In DM, two modes of matching the binlog event are supported (you can only choose one mode from below):
+1. binlog position: the position information of the binlog event 
+    - The binlog position is given by `--binlog-pos` in the command, and the format is `binlog-filename:binlog-pos`, for example, `mysql-bin|000001.000003:3270`.
+    - The format of the binlog filename in DM is not completely consistent with that in the upstream MySQL.
+    - When the replication error occurs, the position can be directly obtained from `failedBinlogPosition` returned by `query-error`.
+2. DDL pattern: the regular expression (only for the DDL statement) matching mode
+    - The DDL pattern is given by `--sql-pattern` in the command, for example, to match ``` ALTER TABLE `db2`.`tbl2` DROP COLUMN `c2` ```, the corresponding regular expression should be ``` ~(?i)ALTER\s+TABLE\s+`db2`.`tbl2`\s+DROP\s+COLUMN\s+`c2` ```.
+    - The regular expression must be prefixed with `~` and cannot contain any common space (you can replace the space with `\s` or `\s+` in the string).
+
+In the scenario of merging and replicating data from sharded tables, if you need DM to automatically select a DDL lock owner to execute the skip/replace operation, then you must use the DDL pattern matching mode because the binlog positions corresponding to the DDL statements on different DM-workers are not logically connected and are hard to confirm.
```

Comment:
*are not logically connected*, is *relationship* better than *connected*?

Reply:
I think maybe you can say: because the binlog positions corresponding to the DDL statements on different DM-workers **have no logical connection** and are hard to confirm. (I think "have no logical connection" is better than "have no logical relationship", how about you?)

Reply:
fine.